### ISSUE TITLE
Create cache for docket pages for anonymous users

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#1365]
This seems to be really hard. You can use the `vary_headers` decorator with Cookies to make it vary on cookies, but since anonymous users have cookies too, this doesn't really work. What you wind up having to do is either checking this in the view or in the template. The template is no place to write code this complicated, so I'm inclined to do it in the view. 

When it's done in the view, there should be a cache that's used for anonymous users that takes into account the GET parameter and the docket ID.

Also, I need to *remove* the current decorators on this view. I'm pretty sure they don't work.

**Comment by mlissner** [comment]
I haven't seen any performance issues on dockets in a long time and I think I've seen cache hits. I think this is more or less solved for now, so closing, but we can reopen it if I'm wrong.

**Comment by mlissner** [comment]
> we can reopen [this] if I'm wrong.

Indeed.

Today I noticed that a docket page was cached after I enabled an alert on it. That wasn't supposed to be happening because logged in users — like somebody creating an alert — weren't supposed to be cached. 

This sent me down a small rabbit hole. The end result is that we have `@vary_on_cookie` enabled for the docket page and that was wrong. The problem with doing that is that you get a cached page for every cookie. That sounds OK, like it'd only affect authenticated users since they're the only ones with cookies, until you realize that, oh shit, everybody has cookies due to Matomo and CSRF. This is literally why this feature was pulled from Django 1.6. 

So instead of creating a cache for anonymous folks only, `vary_on_cookie` does something much worse. It creates a cache for *every damned user*, since every user — authenticated or not — has a cookie of some kind. Guess what? That's worse than not caching at all! 

So....if you want to fix this the trick is probably as described above (cache it in the template and factor in the GET params and whatnot). That's hard. The other solution might be to use Ajax here, but I haven't thought that through completely.

For now, I'm removing the cache. It was worse than nothing because it added both work and bugginess to the view.
